src/main/java/com/google/api/codegen/discovery/MainDiscoveryProviderFactory.java

```diff
@@ -129,10 +120,37 @@ public static DiscoveryProvider defaultCreate(
               new RubySnippetSetRunner<Method>(SnippetSetRunner.SNIPPET_RESOURCE_ROOT))
           .setSnippetFileName(id + "/" + DEFAULT_SNIPPET_FILE)
           .build();
+    }
 
-    } else {
+    // Below is the MVVM pathway.
+    SampleMethodToViewTransformer sampleMethodToViewTransformer = null;
+    TypeNameGenerator typeNameGenerator = null;
+    try {
+      sampleMethodToViewTransformer = SAMPLE_METHOD_TO_VIEW_TRANSFORMER_MAP.get(id).newInstance();
+      typeNameGenerator = TYPE_NAME_GENERATOR_MAP.get(id).newInstance();
+    } catch (InstantiationException | IllegalAccessException e) {
+      throw new RuntimeException(e);
+    }
+    if (sampleMethodToViewTransformer == null || typeNameGenerator == null) {
       throw new NotImplementedException("MainDiscoveryProviderFactory: invalid id \"" + id + "\"");
     }
+
+    ViewModelProvider.Builder viewModelProvider = ViewModelProvider.newBuilder();
```

Comment:
Done, I could've sworn the formatter was doing weird stuff before doing it the other way, but looks like that's not the case anymore haha